src/credentials/FabricTable.h

```diff
@@ -347,9 +347,11 @@ class DLL_EXPORT FabricTable
         virtual ~Delegate() {}
 
         /**
-         * Gets called when a fabric is deleted, such as on FabricTable::Delete().
+         * Gets called when a fabric is changed in a significant way, such as cert being updated or fabric being delete.
```

Comment:
This is too vague; different people will have different interpretations of "significant".

I think we should have clear, well-scoped notifications.  OnFabricDeleted makes sense.  OnFabricNOCChanged makes sense.  Consumers that want to take the same action in both cases can have a helper function both call.

Reply:
Thinking through this, I think this makes sense. There are enough cases where the alteration of the identity at a specific fabric index doesn't actually entail altering fabric-scoped data associated with the previous fabric - that should continue to linger presumably (spec guidance is minimal here) since the admin is swapping out its identity vs. calling remove-and-then-add.

Given that, that likely means there is a distinction between what you would do on fabric-changed vs. fabric-deletion in a lot of impls of this method, justification enough to separate them...

Reply:
Let's just say "When operational credentials are changed".

Reply:
Overall, I really think we should have `OnFabricRemoved` and `OnFabricNOCUpdated`